### PR TITLE
HJ-20 save data categories nested fields

### PR DESCRIPTION
**Pull request by thingscouldbeworse**
Part of HJ-20

### Description Of Changes

remove stipulation that sub fields can't have data categories

### Code Changes

* [ ] remove validation on sub fields

### Steps to Confirm

* [ ] run tests

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Documentation Updated
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: tests/fideslang/test_validation.py [hunk 1]
```diff
@@ -717,43 +717,37 @@ def test_return_all_elements_on_array_field(self):
...
-                    length=None,
-                    return_all_elements=None,
-                    read_only=None,
-                ),
-                fields=[DatasetField(name="nested_field")],
-            )
-        assert_error_message_includes(
-            exc, "Object field 'test_field' cannot have specified data_categories"
+        # Data categories at the object level ARE allowed now
+        DatasetField(
```
Comment (erosselli) [inline tests/fideslang/test_validation.py:721]: should we assign this to a variable and assert that the field both has a category and a subfield? something like `assert field.data_categories == ["user"]` and `assert field.fields == ...`

File: src/fideslang/models.py [hunk 2]
```diff
@@ -471,35 +471,6 @@ def valid_meta(cls, meta_values: Optional[FidesMeta]) -> Optional[FidesMeta]:
...
-
-        if self.fides_meta:
-            declared_data_type = self.fides_meta.data_type
-
-        if fields and declared_data_type:
-            data_type, _ = parse_data_type_string(declared_data_type)
-            if data_type != "object":
-                raise ValueError(
-                    f"The data type '{data_type}' on field '{field_name}' is not compatible with specified sub-fields. Convert to an 'object' field."
-                )
```
Comment (erosselli) [inline src/fideslang/models.py:495]: are we intentionally dropping this data type validation?

File: tests/fideslang/test_validation.py [hunk 3]
```diff
@@ -717,59 +717,61 @@ def test_return_all_elements_on_array_field(self):
...
+                        read_only=None,
+                        data_categories=["user"],
+                    ),
+                )
+            ],
         )
 
+        assert field
+        assert field.data_categories == ["user"]
+        assert field.fields[0].data_categories == ["user"]
```
Comment (thingscouldbeworse) [inline tests/fideslang/test_validation.py:773]: real test here, a nested field can in fact have a category

File: tests/fideslang/test_validation.py [hunk 4]
```diff
@@ -717,59 +717,61 @@ def test_return_all_elements_on_array_field(self):
...
-                    identify=None,
-                    primary_key=False,
-                    data_type="string",
-                    length=None,
-                    return_all_elements=None,
-                    read_only=None,
-                ),
-                fields=[DatasetField(name="nested_field")],
-            )
-        assert_error_message_includes(
```
Comment (thingscouldbeworse) [inline tests/fideslang/test_validation.py:755]: If we want users to add data categories to the "non-terminal" nested fields (middle fields, or top level) then this validation needs to be removed and the reverse of this test is now the case

File: src/fideslang/default_taxonomy/__init__.py [hunk 5]
```diff
@@ -7,8 +7,8 @@
 from .data_uses import DEFAULT_DATA_USES
 from .organizations import DEFAULT_ORGANIZATIONS
 
-sort_data_types = (
-    lambda x: x.parent_key if hasattr(x, "parent_key") and x.parent_key else x.fides_key
+sort_data_types = lambda x: (
```
Comment (thingscouldbeworse) [inline src/fideslang/default_taxonomy/__init__.py:10]: autoformatter

File: src/fideslang/models.py (hunk 2 above)
Comment (thingscouldbeworse) [reply src/fideslang/models.py]: No, added back in now

File: tests/fideslang/test_validation.py (hunk 1 above)
Comment (thingscouldbeworse) [reply tests/fideslang/test_validation.py]: yes good call, added in latest commit

**Review by erosselli — APPROVED** [review]
LGTM! Thanks for making the changes

**Pull request merged**